cmd/kubectl/kubectl.go

```diff
@@ -46,7 +51,48 @@ func main() {
 	logs.InitLogs()
 	defer logs.FlushLogs()
 
+	rest.SetDefaultWarningHandler(newWarningHandler())
+
 	if err := command.Execute(); err != nil {
 		os.Exit(1)
 	}
 }
+
+func newWarningHandler() rest.WarningHandler {
+	dumb := os.Getenv("TERM") == "dumb"
+	_, nocolor := os.LookupEnv("NO_COLOR")
```

Comment:
thanks for `NO_COLOR`

you should consider 
https://github.com/kubernetes-sigs/kind/blob/f277913ed89536b916042b7235e432d65ec81346/pkg/internal/env/term.go#L62-L64
```go
	// On Windows WT_SESSION is set by the modern terminal component.
	// Older terminals have poor support for UTF-8, VT escape codes, etc.
	if GOOS == "windows" && getenv("WT_SESSION") == "" {
```
`WT_SESSION` is set by the new OSS terminal component which has good compatiblity. cmd.exe is pretty bad at anything other than the current codepage (which is localized, not even utf-8 or something).

https://github.com/microsoft/terminal/issues/2256

the bit below in kind about travis is only relevant when doing line rewriting I think.

Reply:
Actually ... IIRC it's possible instead (and not terribly hard) to enable limited proper VT code handling with https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences but to do that you need to `GetStdHandle` / `GetConsoleMode` / `SetConsoleMode`  via kernel32.dll calls.

kind doesn't do this currently we also do line rewriting and other more advanced output in the "smart" terminal mode.

Reply:
EX: https://github.com/containerd/console/blob/8375c3424e4d7b114e8a90a4a40c8e1b40d1d4e6/console_windows.go#L32-L67